src/sign.rs

```diff
@@ -0,0 +1,25 @@
+use std::time::Duration;
+
+/// On some cases like https://github.com/apache/iceberg-rust/issues/506, user may want to
```

Comment:
This comment is not good for lib like `reqsign`.

Yes, https://github.com/apache/iceberg-rust/issues/506 is the motivation for us to add `Sign` but it's not part of the API itself. reqsign has its own design, goal and picture.

I expect to see what's this trait for and how it will be used and implement.

---

In general, `Sign` will be a trait that implement by all services providers supported by `reqsign`, user can implement their own too. It will take `http::request::Parts` and services `Credential` to perform sign over this request.

Some `Sign` impls might have their own logic to sign request without `Credential`, in this cases, user can pass a default `Credential` inside.

Reply:
thank you for the suggestion, i've fixed the docs about it, PTAL